services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -356,4 +362,45 @@ disableForUser zusr tid uid (Public.DisableLegalHoldForUserRequest mPassword) =
       -- TODO: send event at this point (see also: related TODO in this module in
       -- 'approveDevice' and
       -- https://github.com/wireapp/wire-server/pull/802#pullrequestreview-262280386)
-      LegalHoldData.setUserLegalHoldStatus tid uid UserLegalHoldDisabled
+      changeLegalholdStatus tid uid UserLegalHoldDisabled
+
+changeLegalholdStatus :: TeamId -> UserId -> UserLegalHoldStatus -> Galley ()
+changeLegalholdStatus tid uid lhStatus = do
+  LegalHoldData.setUserLegalHoldStatus tid uid lhStatus
+  when (lhStatus == UserLegalHoldPending) $
+    iterateConversations uid (toRange (Proxy @500)) $ \convs ->
+      block1on1s uid (filter ((== One2OneConv) . cnvType) convs)
+  where
+    block1on1s :: UserId -> [Conversation] -> Galley ()
+    block1on1s userLegalhold convs = do
+      let otherUids :: [Qualified UserId] =
+            concatMap (fmap omQualifiedId . cmOthers . cnvMembers) convs
+      ownDomain <- viewFederationDomain
+      let (_remoteUsers, localUids) = partitionRemoteOrLocalIds ownDomain otherUids
+      -- FUTUREWORK: Handle remoteUsers here when federation is implemented
+      for_ (chunksOf 32 localUids) $ \uids -> do
+        teamsOfUser <- Data.usersTeams uids
+        for_ uids $ \user ->
+          case Map.lookup user teamsOfUser of
+            Nothing ->
+              blockConnection user userLegalhold
+            Just teams -> do
+              for_ teams $ \team -> do
+                mMember <- Data.teamMember team user
+                for_ mMember $ \member ->
+                  case member ^. legalHoldStatus of
+                    UserLegalHoldEnabled ->
+                      pure ()
+                    UserLegalHoldDisabled ->
+                      blockConnection user userLegalhold
+                    UserLegalHoldPending ->
+                      blockConnection user userLegalhold
+                    UserLegalHoldNoConsent ->
+                      blockConnection user userLegalhold
```

Comment:
wait...  member is the one who *doesn't* get a LH device, right?  that's the connections of the guy who does.  in that case, the only reason to block if that member is in case `UserLegalHoldNoConsent`.  In all other cases, consent is granted on both sides, and you shouldn't block.